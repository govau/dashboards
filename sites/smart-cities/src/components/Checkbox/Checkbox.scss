@import '~style/main';

$box-size: 16px;

.wrapper {
  position: relative;
}

.input {
  @include hidden;
}

.label {
  padding-left: 25px;
  line-height: $box-size;
  cursor: pointer;
}

// the square for an empty checkbox
.checkbox::before {
  content: '';
  position: absolute;
  top: 0;
  left: 0;
  border: 1px solid $color-grey--500;
  background: $color-white;
  width: $box-size;
  height: $box-size;
  border-radius: 2px;
  margin: 2px; // allow for the focus ring
}

// we need to set the color in JavaScript, so we set 'color' on a span
// and apply it here using currentColor
.input:checked + .label .checkbox::before {
  background: currentColor;
  border: currentColor;
}

// the tick symbol
.input:checked + .label .checkbox::after {
  content: '';
  position: absolute;
  top: 2px;
  left: 7px;
  width: 6px;
  height: 12px;
  border: 2px solid $color-white;
  transform: rotate(45deg);
  backface-visibility: hidden; // required to handle a bug in Chrome
  border-left: transparent;
  border-top: transparent;
}

.input:checked + .label {
  font-weight: 500;
}

:global(body.user-is-tabbing) {
  .input:focus + .label .checkbox::before {
    outline: 2px solid $color-highlight; /* for non-webkit browsers */
    outline: 5px auto -webkit-focus-ring-color;
  }
}
