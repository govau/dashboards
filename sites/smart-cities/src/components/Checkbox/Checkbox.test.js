import React from 'react';
import { shallow } from 'enzyme';
import Checkbox from './Checkbox';

const onChangeMock = jest.fn();

const defaultProps = {
  id: 'the-id',
  label: 'The checkbox label',
  onChange: onChangeMock,
};

it('should match Snapshot', () => {
  const component = shallow(
    <Checkbox {...defaultProps} />
  );

  expect(component.debug()).toMatchSnapshot();
});

it('should apply the id to the input and the label', () => {
  const component = shallow(
    <Checkbox {...defaultProps} />
  );

  expect(component.find('.input').prop('id')).toBe('the-id-checkbox');
  expect(component.find('.label').prop('htmlFor')).toBe('the-id-checkbox');
});

it('applies the passed in color to the checkbox', () => {
  const component = shallow(
    <Checkbox
      {...defaultProps}
      checkedBoxColor="hotpink"
    />
  );

  expect(component.find('.label .checkbox').prop('style').color).toBe('hotpink');
});

it('calls onChange when changed', () => {
  const component = shallow(
    <Checkbox {...defaultProps} />
  );

  component.find('.input').simulate('change', {
    target: {
      checked: true,
    },
  });

  expect(onChangeMock).toHaveBeenCalledTimes(1);
  expect(onChangeMock).toHaveBeenCalledWith(true);
});
