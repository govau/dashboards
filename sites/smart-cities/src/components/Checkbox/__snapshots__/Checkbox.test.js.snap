// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
"<div className=\\"wrapper\\">
  <input className=\\"input\\" type=\\"checkbox\\" id=\\"the-id-checkbox\\" checked={false} onChange={[Function]} />
  <label className=\\"label\\" htmlFor=\\"the-id-checkbox\\">
    <span style={{...}} className=\\"checkbox\\" />
    The checkbox label
  </label>
</div>"
`;
